R/check-up-to-date.R

```diff
@@ -112,7 +112,7 @@ check_up_to_date_nonmem <- function(.mod) {
 
   # check necessary files for changes
   model_file <- get_model_path(.mod)
-  data_file <- get_data_path(.mod)
+  data_file <- get_data_path(.mod, .check_exists = FALSE)
```

Comment:
In the commit message, you say:

> I dont think it should pull from the config here either, as we want to interpret the currently specified path, rather than what it was when the model was submitted.

The model check already takes care of reporting if the data path configured in the model was changed. I don't see a strong reason to change this, and it seems to me that "did the data that was submitted change" is the core question this should be asking. In which case, the value recorded in the config would be the one to consider.

It is unlikely to matter in practice, but in such cases I'd prefer to stick to the current behavior and avoid unnecessary changes.

Reply:
With this change, if you were to change the relative path in the control stream file (but not move the data), you would _now_ receive two warnings: one for the the control stream, and one for the data (cant be found since the relative path is now wrong). Previously, you would only see that the control stream file had changed. 

I dont think this added warning is _that_ necessary, but I do think it would be more informative.

Reply:
> one for the the control stream, and one for the data (cant be found since the relative path is now wrong). Previously, you would only see that the control stream file had changed.

Yes, but, in that case, you're subtlety changing the question, losing the answer to "did that data file that was used to run this model change?"

Reply:
Marking this resolved for the same reasons highlighted [here](https://github.com/metrumresearchgroup/bbr/pull/664#discussion_r1504776552) (i.e. moot point after the refactor).

Reply:
> (i.e. moot point after the refactor).

I agree it's resolved.   I wouldn't classify it as a moot point, any more than any other review comment that is raised and then addressed.